### PR TITLE
fix(FSADT1-1234): adding zap on the CI for verification

### PR DESCRIPTION
**Pull request by paulushcgcj**
# Description

Fixing security issues reported by ZAP

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Further comments

---
Thanks for the PR!

Any successful deployments (not always required) will be available [here](https://nr-forest-client-29-frontend.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-forest-client/actions/workflows/merge-main.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
Any successful deployments (not always required) will be available [here](https://nr-forest-client-29-frontend.apps.silver.devops.gov.bc.ca/)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-forest-client/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-forest-client/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by mishraomp** [comment]
can we remove the backend route @paulushcgcj @DerekRoberts , we don't need it

**Comment by DerekRoberts** [comment]
@mishraomp @paulushcgcj Backend route removed!

**Comment by mishraomp** [comment]
> @mishraomp @paulushcgcj Backend route removed!

I think some changes need to support this, since I dont see caddy reverse prxying request to backend @paulushcgcj ?

**Comment by paulushcgcj** [comment]
> > @mishraomp @paulushcgcj Backend route removed!
> 
> I think some changes need to support this, since I dont see caddy reverse prxying request to backend @paulushcgcj ?

This won't be done right now as it will require more test and some adaptations here and there to accommodate this. For now it stay outside of Caddy

**Comment by mishraomp** [comment]
> > @mishraomp @paulushcgcj Backend route removed!
> 
> Please don't do that, reverting the route for now

may be a ticket in the backlog to align this with QSOS , thoughts?

**Comment by mishraomp** [comment]
do we need this in frontend?
```
    <script>
      window.global = window
      var exports = {}
    </script>
```

**Comment by paulushcgcj** [comment]
This PR will close #850 and #851

**Comment by paulushcgcj** [comment]
This one closes #850

**Comment by paulushcgcj** [comment]
This one closes #851

**Review by mamartinezmejia — APPROVED** [review]

File: backend/src/main/java/ca/bc/gov/app/security/ForestHeadersCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.web.server.ServerHttpSecurity.HeaderSpec;
+import org.springframework.security.config.web.server.ServerHttpSecurity.HeaderSpec.XssProtectionSpec;
+import org.springframework.security.web.server.header.ReferrerPolicyServerHttpHeadersWriter.ReferrerPolicy;
+import org.springframework.security.web.server.header.XFrameOptionsServerHttpHeadersWriter.Mode;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class ForestHeadersCustomizer implements Customizer<HeaderSpec> {
```
[the remark this answers is not in the document]
Comment (paulushcgcj) [reply backend/src/main/java/ca/bc/gov/app/security/ForestHeadersCustomizer.java:17]: I've renamed it in another branch. Will submit as part of the next PR

**Pull request merged**